### PR TITLE
ref(onboarding): Improve vanilla option visibility

### LINKED ISSUE #79397: Make it easier to select vanilla option in framework selection modal
If you select python (or javascript or other "vanilla" option) in onboarding, you are presented with the framework selection modal:
Image

We've received feedback that it's not clear how to select "Nope, I have vanilla" in this step.

We have a "skip" button but it's easy to miss. (or unintuitively named)

We can do two options:
- either we add a generic or vanilla option item to the top of the framework list
- or we rename Skip button to something like "Nope, using vanilla"

We might also consider renaming Configure SDK button because both of them actually continue to configure SDK.

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Before**

https://github.com/user-attachments/assets/18cea2dc-6268-43af-b96d-7c509d1e85fb

**After**

https://github.com/user-attachments/assets/3cc72fa9-fade-4a34-86dd-0fef72ca4fdf

closes https://github.com/getsentry/sentry/issues/79397

### CODE REVIEW

File: static/app/components/onboarding/frameworkSuggestionModal.tsx [hunk 1]
```diff
@@ -238,53 +252,52 @@ export function FrameworkSuggestionModal({
...
                 );
               })}
-            </Frameworks>
+            </PlatformList>
           </StyledPanelBody>
         </StyledPanel>
       </Body>
       <Footer>
-        <Actions>
-          <Button onClick={handleSkip}>{t('Skip')}</Button>
```
Comment (priscilawebdev) [inline static/app/components/onboarding/frameworkSuggestionModal.tsx:278]: we can set the vanilla option as default and remove the 'skip' button, so we don't need to worry about its name.

File: static/app/components/onboarding/frameworkSuggestionModal.tsx [hunk 2]
```diff
@@ -238,53 +252,52 @@ export function FrameworkSuggestionModal({
...
-          <Button
-            priority="primary"
-            onClick={handleConfigure}
-            disabled={!selectedFramework}
-            title={!selectedFramework ? t('Select a framework to configure') : undefined}
```
Comment (priscilawebdev) [inline static/app/components/onboarding/frameworkSuggestionModal.tsx:283]: this validation is not longer needed

File: static/app/components/onboarding/frameworkSuggestionModal.tsx [hunk 3]
```diff
@@ -238,53 +252,52 @@ export function FrameworkSuggestionModal({
         <Description>{languageDescriptions[selectedPlatform.key]}</Description>
         <StyledPanel>
           <StyledPanelBody>
-            <Frameworks>
-              {listEntries.map((framework, index) => {
-                const frameworkCategory =
+            <PlatformList>
```
Comment (priscilawebdev) [inline static/app/components/onboarding/frameworkSuggestionModal.tsx:255]: Renamed components to reflect the new changes

File: static/app/components/onboarding/frameworkSuggestionModal.tsx [hunk 4]
```diff
@@ -238,53 +252,52 @@ export function FrameworkSuggestionModal({
...
-            disabled={!selectedFramework}
-            title={!selectedFramework ? t('Select a framework to configure') : undefined}
-          >
-            {t('Configure SDK')}
-          </Button>
-        </Actions>
+        <Button
+          priority="primary"
+          onClick={
+            selectedFramework?.key === selectedPlatform.key ? handleSkip : handleConfigure
```
Comment (priscilawebdev) [inline static/app/components/onboarding/frameworkSuggestionModal.tsx:296]: In the skip logic, we trigger the `onboarding.select_framework_modal_skip_button_clicked` analytic event. We could update it to reflect the change, or simply leave it as is, even though the skip option is removed.

**Review by vgrozdanic — APPROVED** [review]
Nice 🚀

File: static/app/components/onboarding/frameworkSuggestionModal.tsx [hunk 5]
```diff
@@ -225,7 +225,21 @@ export function FrameworkSuggestionModal({
...
+  const listEntries: PlatformIntegration[] = [
+    ...topFrameworksOrdered,
+    ...otherFrameworksSortedAlphabetically,
+  ];
+
+  const listEntriesWithVanilla: PlatformIntegration[] = [
+    {
+      id: selectedPlatform.key,
+      type: selectedPlatform.type,
+      name: `${selectedPlatform.name} (Vanilla)`,
```
Comment (matejminar) [inline static/app/components/onboarding/frameworkSuggestionModal.tsx:237]:
```suggestion
      name: t('Nope, Vanilla'),
```

**Review by matejminar — APPROVED** [review]

**Pull request merged**